### PR TITLE
Support cmitfb for advanced form with non duplicate case types

### PR DESCRIPTION
**Pull request by emord**
http://manage.dimagi.com/default.asp?223790#1127986

Supports case management in the form builder for advanced forms as long as two of the same case type are not loaded at the same time. (it still half works in that case, but you can't access the second case)

This is fixable now, but once hashtag syntax is released #case/type/prop becomes ambiguous (do you select the first or second case). So I'm going to wait on doing that until we decide on a good way to fix the ambiguity.

@millerdev 

code buddy @proteusvacuum

### CODE REVIEW

File: corehq/apps/app_manager/tests/test_util.py [hunk 1]
```diff
@@ -92,6 +131,16 @@ def assert_has_kv_pairs(self, test_dict, expected_dict):
...
             self.assertEqual(test_dict[key], value)
 
+    def add_advanced_form(self, app, case_types=list(), module_id=None):
+        if module_id is None:
+            module_id = len(app.modules)
+            m = app.add_module(AdvancedModule.new_module('Advanced Module{}'.format(module_id), lang='en'))
+        form = app.new_form(module_id, 'form {}'.format(', '.join(case_types)), lang='en')
+        form.actions.load_update_cases = []
+        for case_type in case_types:
+            form.actions.load_update_cases.append(LoadUpdateAction(case_type=case_type, case_tag='load_' + case_type + '_0'))
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_util.py]: E501 line too long (125 > 115 characters)

File: corehq/apps/app_manager/tests/test_util.py [hunk 2]
```diff
@@ -92,6 +131,16 @@ def assert_has_kv_pairs(self, test_dict, expected_dict):
         for key, value in expected_dict.items():
             self.assertEqual(test_dict[key], value)
 
+    def add_advanced_form(self, app, case_types=list(), module_id=None):
+        if module_id is None:
+            module_id = len(app.modules)
+            m = app.add_module(AdvancedModule.new_module('Advanced Module{}'.format(module_id), lang='en'))
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_util.py]: F841 local variable 'm' is assigned to but never used

File: corehq/apps/app_manager/util.py [hunk 3]
```diff
@@ -346,20 +346,24 @@ def get_session_schema(form):
...
-    # TODO handle advanced modules with more than one case
-    if hasattr(form, 'get_module'):
-        case_type = form.get_module().case_type
-    else:
-        case_type = None
-
-    if case_type:
-        structure["case_id"] = {
+    cases = []
+    # TODO handle advanced modules with more than one use of the same case
```
Comment (snopoke) [inline corehq/apps/app_manager/util.py]: `same case type`

File: corehq/apps/app_manager/util.py [hunk 4]
```diff
@@ -346,20 +346,24 @@ def get_session_schema(form):
...
+    if form.doc_type == "AdvancedForm":
+        for action in form.actions.load_update_cases:
+            cases.append((action.case_type, action.case_session_var))
+    elif hasattr(form, 'get_module'):
+        if form.get_module().case_type:
+            cases.append((form.get_module().case_type, 'case_id'))
```
Comment (snopoke) [inline corehq/apps/app_manager/util.py]: These session vars can get changed at app build time depending on the workflow. This is how we do it for cloudcare: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/cloudcare/touchforms_api.py#L118-L124

**Comment by emord** [comment]
Closing this as it was out of scope, and cmitfb has changed since then

**Pull request closed without merging**